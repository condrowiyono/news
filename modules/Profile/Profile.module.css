.container {
  padding: 16px 24px;
  background: #FFFFFF;
  border-bottom: 1px solid #E5E9EC;
  width: calc(100%-48px);
}

.profile {
  display: flex;
}

.identityContainer {
  margin-left: 16px;
}

.stats {
  display: flex;
  gap: 40px;
}

.info {
  margin-bottom: 10px;
}

.label {
  font-weight: 400;
  font-size: 11px;
  line-height: 140%;
  color: #797d7f;
}

.content {
  font-style: normal;
  font-weight: 900;
  font-size: 16px;
  line-height: 130%;
  color: #14171A;
}

.description {
  font-style: normal;
  font-weight: 400;
  font-size: 14px;
  line-height: 160%;
  color: #14171A;
  margin-bottom: 16px;
}

@media only screen and (min-width: 480px) {
  .container {
    width: 375px;
  }
}