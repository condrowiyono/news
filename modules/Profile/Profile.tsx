import Button from "@components/Button";
import ImageServer from "@components/Image";
import style from "./Profile.module.css";
import { formatNumber } from "@utils/number";

const Profile = ({
  name,
  num_follower,
  num_following,
  profile_picture,
  description,
}: IProfile) => {
  return (
    <div className={style.container}>
      <div className={style.profile}>
        <div>
          <ImageServer
            src={profile_picture}
            objectFit="cover"
            width={80}
            height={80}
            alt="profile-picture"
            style={{ borderRadius: "50%" }}
          />
        </div>
        <div className={style.identityContainer}>
          <div className={style.info}>
            <div className={style.label}>Verified Creator</div>
            <div className={style.content}>{name}</div>
          </div>
          <div className={style.stats}>
            <div className={style.info}>
              <div className={style.label}>Following</div>
              <div className={style.content}>{formatNumber(num_following)}</div>
            </div>
            <div className={style.info}>
              <div className={style.label}>Follower</div>
              <div className={style.content}>
                {formatNumber(num_follower, true)}
              </div>
            </div>
          </div>
        </div>
      </div>
      <div className={style.description}>{description}</div>
      <Button />
    </div>
  );
};

export default Profile;
