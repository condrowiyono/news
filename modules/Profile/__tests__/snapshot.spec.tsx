import { render } from "@testing-library/react";
import Profile from "@modules/Profile";
import { profile } from "./Profile.spec";

it("renders Profile unchanged", () => {
  const { container } = render(<Profile {...profile} />);
  expect(container).toMatchSnapshot();
});
