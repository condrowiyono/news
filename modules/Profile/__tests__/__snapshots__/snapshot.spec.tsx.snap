// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Profile unchanged 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="profile"
    >
      <div>
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2780%27%20height=%2780%27/%3e"
              style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
            />
          </span>
          <img
            alt="profile-picture"
            data-nimg="intrinsic"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="border-radius: 50%; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover; background-size: cover; background-position: 0% 0%; filter: blur(20px); background-image: url(data:image/gif;base64,R0lGODlhAQABAPAAAL29vf///yH5BAAAAAAALAAAAAABAAEAAAICRAEAOw==);"
          />
          <noscript />
        </span>
      </div>
      <div
        class="identityContainer"
      >
        <div
          class="info"
        >
          <div
            class="label"
          >
            Verified Creator
          </div>
          <div
            class="content"
          >
            Samantha William
          </div>
        </div>
        <div
          class="stats"
        >
          <div
            class="info"
          >
            <div
              class="label"
            >
              Following
            </div>
            <div
              class="content"
            >
              9,998
            </div>
          </div>
          <div
            class="info"
          >
            <div
              class="label"
            >
              Follower
            </div>
            <div
              class="content"
            >
              99.8rb
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="description"
    >
      Yuk follow aku biar kita streaming game bareng setiap hari senin & jumat jam 10 AM - 3 PM. Jangan lupa follow IG aku juga ya @Sam_will
    </div>
    <button
      class="button"
      data-testid="button"
    >
      Follow
    </button>
  </div>
</div>
`;
